File: euler/tyama_euler29.rb
#!/usr/bin/ruby
n=gets.to_i
h={}
2.step(n){|a0|
	a=a0*a0
	2.step(n){|b|
		h[a]=1
		a*=a0
	}
}
p h.size
__END__
100

File: euler/tyama_euler29.py
#!/usr/bin/python
n=int(raw_input())
h=set()
for a0 in xrange(2,n+1):
	a=a0*a0
	for b in xrange(2,n+1):
		h.add(a)
		a*=a0
print(len(h))
# 100